core/templates/components/copy-exploration-url/copy-exploration-url.component.ts

```diff
@@ -0,0 +1,28 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
```

Comment:
Can we have a more general component that can take any url as an input, allows it to be copied and show the tooltip for it (rather that just having a separate component for tool-tip) ?

Reply:
No, I tried that but because both the URLs are generated by a different logic, and also the copy button has different logics for copying the URL, so there can't be a common component for that.

Reply:
The URL can be generated separately and then only used in the common component. What is the difference between the copy logic?

Reply:
Yes, I am done with the changes. I have passed the URLs to the child component and done some more changes to generalize the component! PTAL.

Reply:
Yes, I am done with the changes. I have passed the URLs to the child component and done some more changes to generalize the component! PTAL.